lib/shared/types.js

```diff
@@ -189,11 +189,44 @@ module.exports = {};
  * @property {number} warningCount Number of warnings for the result.
  * @property {number} fixableErrorCount Number of fixable errors for the result.
  * @property {number} fixableWarningCount Number of fixable warnings for the result.
+ * @property {Stats} Stats The performance statistics collected with the `stats` flag.
  * @property {string} [source] The source code of the file that was linted.
  * @property {string} [output] The source code of the file that was linted, with as many fixes applied as possible.
  * @property {DeprecatedRuleInfo[]} usedDeprecatedRules The list of used deprecated rules.
  */
 
+/**
+ * Performance statistics
+ * @typedef {Object} Stats
+ * @property {number} fixPasses The number of times ESLint has applied at least one fix after linting.
+ * @property {Times} times The times spent on (parsing, fixing, linting) a file.
+ */
+
+/**
+ * Performance Times for each ESLint pass
+ * @typedef {TimePass[]} Time passes
+ */
+
+/**
+ * @typedef {Object} TimePass
+ * @property {ParseTime} parse The parse object containing all parse time information.
+ * @property {Record<string, RuleTime>} rules The rules object containing all lint time information for each rule.
+ * @property {FixTime} fix The parse object containing all fix time information.
+ * @property {string} total The total time that is spent on (parsing, fixing, linting) a file.
+ */
+/**
+ * @typedef {Object} ParseTime
+ * @property {number} total The total time that is spent when parsing a file.
+ */
+/**
+ * @typedef {Object} RuleTime
+ * @property {number} total The total time that is spent on a rule.
+ */
+/**
+ * @typedef {Object} FixTime
+ * @property {number} total The total time that is spent on a fix pass.
```

Comment:
```suggestion
 * @property {number} total The total time that is spent on applying fixes to the code.
```

"fix pass" may sound like the total time spent in a pass, while this is specifically time spent on applying fixes.